Docs: Add contributing guidelines to branch 6.0 (#548)
mautic/user-documentation: .github/CONTRIBUTING.md

# Contributing to Mautic user documentation

Contributions are always welcome, no matter how large or small, or at whatever skill level you are. Before contributing, please read the [Code of Conduct](https://mautic.org/code-of-conduct/) and follow the directions in this guide.

---

## Table of contents

- [Communication expectation](#communication-expectation)
- [Issues](#issues)
- [Pull requests (PRs)](#pull-requests-prs)
  - [Before submitting a PR](#before-submitting-a-pr)
  - [Submitting a PR](#submitting-a-pr)
  - [After submitting a PR](#after-submitting-a-pr)
- [Contributing workflow](#contributing-workflow)
  - [Forking the repository](#forking-the-repository)
  - [Clone the repository](#clone-the-repository)
  - [Create a new branch](#create-a-new-branch)
    - [Ensure the correct base branch](#ensure-the-correct-base-branch)
    - [Ways to create a new branch](#ways-to-create-a-new-branch)
  - [Push changes to the remote repository](#push-changes-to-the-remote-repository)
  - [Create a PR](#create-a-pr)
    - [Git cherry-pick](#git-cherry-pick)
- [Getting started](#getting-started)
  - [1. On GitHub](#1-on-github)

>   - [2. GitHub Codespaces](#2-github-codespaces)

Reviewer: We should add these terms to the accept.txt - made a suggestion for review.

- [Working with links](#working-with-links)
  - [Create a new link](#create-a-new-link)
  - [Check broken links](#check-broken-links)
- [Working with Vale](#working-with-vale)
- [Updating contents and UI images](#updating-contents-and-ui-images)
- [Credit](#credit)

---

## Communication expectation

1. Always leave a detailed description in the pull request (PR). Leave nothing ambiguous for the reviewers.
2. Provide screenshots for visual changes.
3. Always review your code first. Be sure to run the project locally and test it thoroughly before requesting a review.
4. Communicate in the GitHub repository first before Slack. Whether it's in the issue or the PR, keeping the lines of communication open and visible to everyone on the team helps everyone around you.

## Issues

- When you contribute to the project for the first time, please consider checking the [good first issue](https://github.com/mautic/user-documentation/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22good%20first%20issue%22) or [help wanted](https://github.com/mautic/user-documentation/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22help%20wanted%22) labels.

> - If you wish to work on an open issue, please comment on the issue, and a maintainer will assign it to you.

Reviewer: ```suggestion
- If you wish to work on an open issue, please comment on the issue requesting a maintainer to assign you.
```

  If an issue isn't assigned, it's assumed to be available for anyone to work on. So, ensure that you're assigned to an issue **before** beginning work to avoid conflicts.

>   **Note:** Please don't ask maintainers to assign you to another issue before you have finished working on yours and created a PR.

Reviewer: Let's use the Note admonition here to make it look pretty! :)

> - Please don't ask maintainers to assign you to an issue that already has someone assigned to it. If the assigned issue hasn't been addressed for a while and you're interested in working on it, leave a comment to ask about its status and progress.

Reviewer: ```suggestion
- Please don't ask maintainers to assign you to an issue that already has someone assigned to it. If there's no recent update on an issue for a while and you're interested in working on it, leave a comment to ask about its status and progress.
```

- Did you spot a typo, missing instructions, or have an idea for enhancing the Mautic User Documentation? You can [create an issue](https://github.com/mautic/user-documentation/issues/new/choose) to address it.

  However, the Education Team needs to triage the issue before you can work on it. If you wish to work on the issue you submitted, please inform and tag the `@mautic/education-team-leaders` in the comment.

## Pull requests (PRs)

> PRs are always welcome. However, before working on changes, you must ensure that **you are assigned** to an existing issue and **link your work to the issue in your PR**.

Reviewer: ```suggestion
PRs are always welcome. However, before working on changes, wait to be **assigned** to an existing issue and ensure you always **link your work to the issue in your PR**.
```

3. Keep your branch up to date while waiting for review.
4. Respond and address the reviewer's feedback. Please don't request a review until you've addressed all feedback.

## Contributing workflow

In this section, you can find the contributing workflow and best practices for contributing to this project.

### Forking the repository

Forking this repository is the first step you need to take before proceeding. Forking means making a copy of a repository to your GitHub account.

To fork this repository:

1. In the main page of [mautic/user-documentation](https://github.com/mautic/user-documentation), click the 'Fork' button at the top.

   ![Fork button on GitHub](../assets/images/fork_button_github.png)

2. Choose your username in the 'Owner *' dropdown menu. **Don't select an organization here. Always choose your personal account**; otherwise, the team can't collaborate or fix issues in your PR.

   ![Choose fork owner on GitHub](../assets/images/choose_fork_owner_github.png)

> 3. Uncheck the option of 'Copy the DEFAULT-BRANCH-NAME branch only' because later on, you want to be able to clone multiple branches.

Reviewer: ```suggestion
3. Remove the check in the option of 'Copy the DEFAULT-BRANCH-NAME branch only' because later on, you want to be able to clone multiple branches.
```

4. Click the green 'Create fork' button at the bottom.

>    ![An uncheck state option to copy only default branch and a create fork button on GitHub](../assets/images/uncheck_option_and_create_fork_button_github.png)

Reviewer: ```suggestion
   ![An empty checkbox state option to copy only default branch and a create fork button on GitHub](../assets/images/uncheck_option_and_create_fork_button_github.png)
```

Follow the steps below to clone your forked repository:

1. Click your avatar on the top right.
2. Click 'Repositories'.

   ![Repositories option from a dropdown menu on GitHub](../assets/images/repositories_option_github.png)

3. Open your forked user-documentation repository. The URL should have your username. For example: `https://github.com/YOUR-GITHUB-USERNAME/user-documentation`.
4. Click the green 'Code' button on top.
5. Select 'HTTPS' and copy the URL.

   ![Highlight of code button, copy symbol, and HTTPS tab on GitHub](../assets/images/code_button_https_tab_github.png)

6. In your terminal, go to your local directory where you want to save this project.
7. Run the `git clone` command specifying the branch and folder name, and hit Enter:

   ```bash
   git clone --branch BRANCH-NAME https://github.com/YOUR-GITHUB-USERNAME/user-documentation FOLDER-NAME
   ```

   Here are some examples:

   ```bash
   git clone --branch 6.0 https://github.com/YOUR-GITHUB-USERNAME/user-documentation user-docs-6
   git clone --branch 7.0 https://github.com/YOUR-GITHUB-USERNAME/user-documentation user-docs-7
   ```

### Create a new branch

> Before making changes, ensure that you create a new branch and work on it. You don't want to directly work on the default — such as `main` — or any other base branch, because you won't be able to work on lots of things at the same time. If you make all those changes on one branch, you can't separate them and merge only one change at a time.

Reviewer: ```suggestion
Before making changes, ensure that you create a new branch and work on it. You don't want to directly work on the default - such as `main` - or any other base branch, because you won't be able to work on lots of things at the same time. If you make all those changes on one branch, you can't separate them and merge only one change at a time.
```
